docs/components/material.md

```diff
@@ -498,44 +504,231 @@ void main() {
 
 varying vec2 vUv;
 uniform vec3 color;
-uniform float time;
+uniform float time1000;
 
 void main() {
   // Use sin(time), which curves between 0 and 1 over time,
   // to determine the mix of two colors:
   //    (a) Dynamic color where 'R' and 'B' channels come
   //        from a modulus of the UV coordinates.
   //    (b) Base color.
-  //
+  // 
   // The color itself is a vec4 containing RGBA values 0-1.
   gl_FragColor = mix(
     vec4(mod(vUv , 0.05) * 20.0, 1.0, 1.0),
     vec4(color, 1.0),
-    sin(time)
+    sin(time1000 / 1000.0) // since A-Frame time is in milliseconds
   );
 }
 ```
 
+To use these vertex and fragment shaders, 
+after reading them into strings `vertexShader` and `fragmentShader`,
+we simply register our custom shader with A-Frame:
+
+```js
+// shader-grid-glitch.js
+
+AFRAME.registerShader('grid-glitch', {
+  schema: {
+    color: {type: 'color', is: 'uniform'},
+    time1000: {type: 'time', is: 'uniform'}
+  },
+
+  vertexShader: vertexShader,
+  fragmentShader: fragmentShader
+});
+```
+
 Finally, here is the HTML markup to put it all together:
 
 ```html
 <!-- index.html -->
 
 <a-scene>
-  <a-sphere material-grid-glitch="color: blue;"
+  <a-sphere material="shader:grid-glitch; color: blue;"
             radius="0.5"
             position="0 1.5 -2">
   </a-sphere>
 </a-scene>
 ```
 
-* [Live demo](https://aframe-simple-shader.glitch.me/)
-* [Remix this on Glitch](https://glitch.com/edit/#!/aframe-simple-shader)
+* [Live demo](https://aframe-simpler-shader.glitch.me/)
+* [Remix this on Glitch](https://glitch.com/edit/#!/aframe-simpler-shader)
 
 ![5093034e-97f2-40dc-8cb9-28ca75bfd75b-8043-00000dbc2e00268d](https://cloud.githubusercontent.com/assets/1848368/24825516/abb98abe-1bd4-11e7-8262-93d3efb6056f.gif)
 
 ***
 
-For a more advanced example, [try realtime vertex displacement](https://glitch.com/edit/#!/aframe-displacement-shader).
+For a more advanced example, [try realtime vertex displacement](https://glitch.com/edit/#!/aframe-displacement-registershader).
 
 ![b19320eb-802a-462a-afcd-3d0dd9480aee-861-000004c2a8504498](https://cloud.githubusercontent.com/assets/1848368/24825518/b52e5bf6-1bd4-11e7-8eb2-9a9c1ff82ce9.gif)
+
+## Using a Custom Shader and Component Together
+
+Let's take the realtime vertex displacement shader example above, and
+add the capability to apply an offset based upon the camera's position.
+
+We declare that offset as a uniform vec3 value `myOffset`:
+
+```js
+AFRAME.registerShader('displacement-offset', {
+  schema: {
+    time1000: {type:'time', is:'uniform'},
+    myOffset: {type:'vec3', is:'uniform'}
+  },
+  vertexShader: vertexShader,
+  fragmentShader: fragmentShader
+});
+```
+
+that the vertex shader uses:
+
+```glsl
+// vertex.glsl
+
+#pragma glslify: pnoise3 = require(glsl-noise/periodic/3d)
```

Comment:
I probably wouldn't go to the trouble of explaining how to set it up here, but worth noting somewhere that:

> This example uses [Browserify](http://browserify.org/) and [glslify](https://github.com/glslify/glslify) to include  [glsl-noise](https://github.com/hughsk/glsl-noise) as a dependency of our shader.

(similar options are available for rollup etc.)

Reply:
done